src/DDTrace/OpenTracer/Scope.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace DDTrace\OpenTracer;
+
+use DDTrace\Contracts\Scope as ScopeInterface;
+use OpenTracing\Scope as OTScope;
+
+final class Scope implements OTScope
+{
+    /**
+     * @var ScopeInterface
+     */
+    private $scope;
+
+    /**
+     * @var Span
+     */
+    private $span;
+
+    /**
+     * @param ScopeInterface $scope
+     */
+    public function __construct(ScopeInterface $scope)
+    {
+        $this->scope = $scope;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function close()
+    {
+        $this->scope->close();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSpan()
+    {
+        if (isset($this->span)) {
```

Comment:
Do we need this optimization? Just to play safe I would always ask the scope for the current span, one less thing to keep in sync :)

Reply:
My thought on this was that every time you call `getSpan()` here, you'll always get the same instance of the wrapper (just like the unwrapped version). Without this check, the internal call to `$this->scope->getSpan()` will always be the same instance, but the wrapper instance that is returned will always be different.

It's not a big deal either way and I can totally remove the check if you like the other way better. :)

Reply:
This actually makes a lot of sense. Thanks for clarifying!